name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/ibflex/IBFlexStatementExtractor.java

```diff
@@ -263,16 +276,35 @@ else if (this.ibAccountCurrency.equals(s.getCurrencyCode()))
             setAmount(element, transaction, amount);
 
             // Set note
-            transaction.setNote(element.getAttribute("description"));
+            if (!transaction.getType().equals(AccountTransaction.Type.DIVIDENDS))
+                transaction.setNote(element.getAttribute("description"));
 
             // Add Trade-ID note if available
-            if (!element.getAttribute("tradeID").isEmpty() && !element.getAttribute("tradeID").equals("N/A"))
-                transaction.setNote("Trade-ID: " + element.getAttribute("tradeID") + " | " + transaction.getNote());
+            if (transaction.getNote() == null)
+            {
+                if (!element.getAttribute("tradeID").isEmpty() && !element.getAttribute("tradeID").equals("N/A"))
+                    transaction.setNote("Trade-ID: " + element.getAttribute("tradeID"));
+            }
+            else
+            {
+                if (!element.getAttribute("tradeID").isEmpty() && !element.getAttribute("tradeID").equals("N/A"))
+                    transaction.setNote("Trade-ID: " + element.getAttribute("tradeID") + " | " + transaction.getNote());
```

Comment:
@Nirus2000: Another note: it's certainly useful to record tradeID/transactionID to cross-reference to the original datasource in case it ever needed, but I don't think that's the most useful/most frequent information a user may want to see. Thus, it would make sense to put it at the end of the note, not at the beginning.

Reply:
This is a personal view and you can change the note yourself at any time. 
The ID's should be at the beginning to keep the column widths small for country descriptions.
You can identify what it is by the transaction type.

Example:
"Transaction-ID: 12345679 | USD IBKR MANAGED SECURITIES (SYEP) INTEREST FOR NOV-2022"
Transaction type is interest

Reply:
> This is a personal view and you can change the note yourself at any time.

Sure, and I'm not calling for an immediate change, just providing feedback and discussion of the work you're doing, to both show appreciation and in the hope to make further small improvements.

> The ID's should be at the beginning to keep the column widths small for country descriptions.

Hmm, I'm not sure I understand this. Let me elaborate my point. If user selects to show the Note column, it's width certainly will be limited (by other columns or physical screen width), so given your example, a user could see for example "Transaction-ID: 12345679 | USD". I'm not sure that seeing a transaction ID per se would be useful to end user most of the time. IMHO, it would be better to show "USD IBKR MANAGED SEC", much more informative.